sdk/src/main/scala/com/horizen/api/http/SidechainSubmitterApiRoute.scala

```diff
@@ -82,17 +80,17 @@ case class SidechainSubmitterApiRoute(override val settings: RESTApiSettings, ce
     ApiResponseUtil.toResponse(RespSubmitterOk)
   }
 
-  def getSchnorrPublicKeyHash: Route = (post & path("getSchnorrPublicKeyHash")) {
-    entity(as[ReqGetSchnorrPublicKeyHash]) { body =>
-      try {
-        val schnorrPublicKey: SchnorrProposition = new SchnorrProposition(BytesUtils.fromHexString(body.schnorrPublicKey))
-        ApiResponseUtil.toResponse(
-          RespHashSchnorrPublicKey(
-            BytesUtils.toHexString(schnorrPublicKey.getHash)
```

Comment:
Changes in the SidechainState are missing.
The code in the API was updated: instead of `SchnorrProposition.getHash` we now use `CryptoLibProvider.thresholdSignatureCircuitWithKeyRotation.getMsgToSignForSigningKeyUpdate`.
But in the State during verification we still use the old `SchnorrProposition.getHash`.
Check and update the following methods:
1) SidechainState.validateWithWithdrawalEpoch
2) SidechainStateState.getKeyRotationTransaction

Moreover, let's remove SchnorrProposition.getHash method at all, it will clearly show now and during Account model integration all places where we need to change the code.

Reply:
Update also the **SidechainStateTest** accordingly (e.g. _getKeyRotationTransaction()_, _keyRotationTransactionTest()_

Reply:
I updated SidechainStateTest, but there are a lot of things to change there in order to make it work. Could you please checkout the branch SDK-681, pull the latest changes, run SidechainStateTest and see what can we modify?